src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ProcessTests.cs

```diff
@@ -145,7 +145,137 @@ public async Task MultiProcessIdentificationTest(DiagnosticPortConnectionMode mo
+                {
+                    pids.Add(processIdentifier.Pid);
+                    uids.Add(processIdentifier.Uid);
+                    names.Add(processIdentifier.Name);
+                }
+
+#if NET5_0_OR_GREATER
+                // CHECK 1: Get response for processes using PID, UID, and Name and check for consistency
+
+                List<ProcessInfo> processInfoQueriesCheck1 = new List<ProcessInfo>();
+
+                processInfoQueriesCheck1.Add(await apiClient.GetProcessAsync(pid: pids[0], uid: null, name: null, System.Threading.CancellationToken.None));
+                processInfoQueriesCheck1.Add(await apiClient.GetProcessAsync(pid: null, uid: uids[0], name: null, System.Threading.CancellationToken.None));
+                //processInfoQueriesCheck1.Add(await apiClient.GetProcessAsync(pid: null, uid: null, name: names[0], System.Threading.CancellationToken.None));
+
+                VerifyProcessInfoEquality(processInfoQueriesCheck1);
+#endif
+                // CHECK 2: Get response for requests using PID | PID and UID | PID, UID, and Name and check for consistency
+
+                List<ProcessInfo> processInfoQueriesCheck2 = new List<ProcessInfo>();
+
+                processInfoQueriesCheck2.Add(await apiClient.GetProcessAsync(pid: pids[0], uid: null, name: null, System.Threading.CancellationToken.None));
+                processInfoQueriesCheck2.Add(await apiClient.GetProcessAsync(pid: pids[0], uid: uids[0], name: null, System.Threading.CancellationToken.None));
+                processInfoQueriesCheck2.Add(await apiClient.GetProcessAsync(pid: pids[0], uid: uids[0], name: names[0], System.Threading.CancellationToken.None));
+
+                VerifyProcessInfoEquality(processInfoQueriesCheck2);
+
+                // CHECK 3: Get response for processes using PID and an unassociated (randomly generated) UID and ensure the proper exception is thrown
+
+                await Assert.ThrowsAsync<ValidationProblemDetailsException>(async () => await apiClient.GetProcessAsync(pid: pids[0], uid: Guid.NewGuid(), name: null, System.Threading.CancellationToken.None));
+
+                // CHECK 4: Get response for processes using invalid PID, UID, or Name and ensure the proper exception is thrown
+
+                await Assert.ThrowsAsync<ValidationProblemDetailsException>(async () => await apiClient.GetProcessAsync(pid: -1, uid: null, name: null, System.Threading.CancellationToken.None));
+                await Assert.ThrowsAsync<ValidationProblemDetailsException>(async () => await apiClient.GetProcessAsync(pid: null, uid: Guid.NewGuid(), name: null, System.Threading.CancellationToken.None));
+                await Assert.ThrowsAsync<ValidationProblemDetailsException>(async () => await apiClient.GetProcessAsync(pid: null, uid: null, name: "", System.Threading.CancellationToken.None)); // Are we safe to assume that processes will not have an empty name?
```

Comment:
> // Are we safe to assume that processes will not have an empty name?

Yes, I think that's a reasonable assumption. We are also unlike to ever attempt to launch a process with an empty name, even if it happens to be supported.

However, you should check in dotnet-monitor if ASP.NET Core translates that to an empty string or null string as an argument on the action method.